monix-execution/shared/src/main/scala/monix/execution/misc/Local.scala

```diff
@@ -148,14 +150,17 @@ object Local extends LocalCompanionDeprecated {
     def isolate(f: Tree): Tree =
       c.abort(c.macroApplication.pos, "Macro no longer implemented!")
 
-    def localLetCurrentIf(b: Tree)(f: Tree): Tree = {
+    def localLetCurrentIf(b: Tree, isolate: Tree)(f: Tree): Tree = {
       val Local = symbolOf[Local[_]].companion
       val CanBindLocals = symbolOf[CanBindLocals[_]].companion
 
       resetTree(
         q"""if (!$b) { $f } else {
            import $CanBindLocals.Implicits.synchronousAsDefault
-           $Local.isolate($f)
+           if ($isolate)
+             $Local.isolate($f)
+           else
+             $Local.bind($Local.getContext())($f)
```

Comment:
is it necessary to `bind` if we don't isolate?

Reply:
Its the only method I could find, is there another one I should be using?

Reply:
I'm wondering if it can be old implementation of this method - did you encounter any problems?

I think it will be already current context (`Local.getContext()`) and the runloop should take care of preserving it across async boundaries

Reply:
So I am just doing `$f` now and the unit test seems to pass so I am using this now.